### ISSUE TITLE
BoundsGroup bounds min & max have incorrect cardinality

### ISSUE THREAD

**Issue by mbjones** [NCEAS/eml#127]
---

Author Name: **Matthew Brooke** (Matthew Brooke)
Original Redmine Issue: 1149, https://projects.ecoinformatics.org/ecoinfo/issues/1149
Original Date: 2003-09-17
Original Assignee: Matt Jones

---

BoundsGroup bounds min & max have incorrect cardinality. This can be seen in the
following places:

NumericDomainType:

numericDomain/bounds/minimum & numericDomain/bounds/maximum currently have
incorrect cardinality - should be 0..1 instead of 0..infinity

DateTimeDomainType:

dateTimeDomain/bounds/minimum & dateTimeDomain/bounds/maximum currently have
incorrect cardinality - should be 0..1 instead of 0..infinity

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Matt Jones** (Matt Jones)
Original Date: 2003-10-28T20:36:16Z

---

Changed the content model to fix this problem.  Now the model allows either
minimum, maximum, or both, as described in DTD syntax:

   bounds(minimum | maximum | (minimum, maximum))

This was done for both BoundsGroup (used in numeric domains) and BoundsDateGroup
(used in dateTime domains).

FIXED.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Matt Jones** (Matt Jones)
Original Date: 2003-10-30T23:59:36Z

---

When running the tests I found that this content model is rejected.  Maybe its
because it is not deterministic according to the xml spec?  Need to fix this
model before releasing the 2.0.1 version.

**Comment by mbjones** [comment]
---

Original Redmine Comment
Author Name: **Matt Jones** (Matt Jones)
Original Date: 2003-11-06T20:24:47Z

---

Checked on the bounds model -- turns out that the model I proposed before is not
deterministic according to XML (without look ahead) and so is illegal.  SO I
changed the bounds group to this model:
 
bounds (min?, max?)

So at least now they are not repeatable.   Plus date bounds has a different type
now so that the documentation and type for the bounds field can differ from the
numeric bounds.

FIXED.
